example/zipIterator/src/zipIterator-main.cpp

```diff
@@ -0,0 +1,245 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <vikunja/transform/transform.hpp>
+#include <vikunja/mem/iterator/ZipIterator.hpp>
+
+#include <alpaka/alpaka.hpp>
+
+#include <iostream>
+#include <variant>
+
+template<std::size_t I = 0, typename FuncT, typename... Tp>
+inline typename std::enable_if<I == sizeof...(Tp), void>::type forEach(std::tuple<Tp...> &, FuncT) // Unused arguments are given no names
+{
+}
+
+template<std::size_t I = 0, typename FuncT, typename... Tp>
+inline typename std::enable_if<I < sizeof...(Tp), void>::type forEach(std::tuple<Tp...>& t, FuncT f)
+{
+    f(std::get<I>(t));
+    forEach<I + 1, FuncT, Tp...>(t, f);
+}
+
+template<typename IteratorTuplePtr>
+void printTuple(IteratorTuplePtr tuple)
+{
+    std::cout << "tuple(";
+    int index = 0;
+    int tupleSize = std::tuple_size<IteratorTuplePtr>{};
+    forEach(tuple, [&index, tupleSize](auto &x) { std::cout << x << (++index < tupleSize ? ", " : ""); });
+    std::cout << ")";
+}
+
+int main()
+{
+    // Define the accelerator here. Must be one of the enabled accelerators.
+    using TAcc = alpaka::AccCpuSerial<alpaka::DimInt<3u>, std::uint64_t>;
+
+    // Types of the data that will be reduced
+    using TRed = uint64_t;
+    using TRedChar = char;
+    using TRedDouble = double;
+
+    // Alpaka index type
+    using Idx = alpaka::Idx<TAcc>;
+    // Alpaka dimension type
+    using Dim = alpaka::Dim<TAcc>;
+    // Type of the extent vector
+    using Vec = alpaka::Vec<Dim, Idx>;
+    // Find the index of the CUDA blockIdx.x component. Alpaka somehow reverses
+    // these, i.e. the x component of cuda is always the last value in the vector
+    constexpr Idx xIndex = Dim::value - 1u;
+    // number of elements to reduce
+    const Idx n = static_cast<Idx>(10);
+    // create extent
+    Vec extent(Vec::all(static_cast<Idx>(1)));
+    extent[xIndex] = n;
+
+    // define device, platform, and queue types.
+    using DevAcc = alpaka::Dev<TAcc>;
+    using PltfAcc = alpaka::Pltf<DevAcc>;
+    // using QueueAcc = alpaka::test::queue::DefaultQueue<alpaka::Dev<TAcc>>;
+    using PltfHost = alpaka::PltfCpu;
+    using DevHost = alpaka::Dev<PltfHost>;
+    using QueueAcc = alpaka::Queue<TAcc, alpaka::Blocking>;
+    using QueueHost = alpaka::QueueCpuBlocking;
+
+    // Get the host device.
+    DevHost devHost(alpaka::getDevByIdx<PltfHost>(0u));
+    // Get a queue on the host device.
+    QueueHost queueHost(devHost);
+    // Select a device to execute on.
+    DevAcc devAcc(alpaka::getDevByIdx<PltfAcc>(0u));
+    // Get a queue on the accelerator device.
+    QueueAcc queueAcc(devAcc);
+
+    // allocate memory both on host and device.
+    auto deviceMem(alpaka::allocBuf<TRed, Idx>(devAcc, extent));
+    auto hostMem(alpaka::allocBuf<TRed, Idx>(devHost, extent));
+    // Fill memory on host with numbers from 0...n-1.
+    TRed* hostNative = alpaka::getPtrNative(hostMem);
```

Comment:
The native host memory type needs to be the same type, like the return type of the functor. In your case, it needs to be a `std::tuple<std::uint64_t, char, double>`.